### PR TITLE
feat(facade): add range to-string api & fix data-validation api

### PR DESCRIPTION
**Pull request by weird94**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/sheets-data-validation/src/facade/f-worksheet.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export interface IFWorksheetDataValidationMixin {
 export class FWorksheetDataValidationMixin extends FWorksheet implements IFWorksheetDataValidationMixin {
     override getDataValidations(): FDataValidation[] {
         const dataValidationModel = this._injector.get(DataValidationModel);
-        return dataValidationModel.getRules(this._workbook.getUnitId(), this._worksheet.getSheetId()).map((rule) => new FDataValidation(rule));
+        return dataValidationModel.getRules(this._workbook.getUnitId(), this._worksheet.getSheetId()).map((rule) => new FDataValidation(rule, this._worksheet, this._injector));
```
Comment (wzhudev) [inline packages/sheets-data-validation/src/facade/f-worksheet.ts:39]: 应该不需要手动调用 new ，试试

```ts
this._injector.createInstance(rule, this._worksheet)
```

**Review by wzhudev — APPROVED** [review]

File: packages/sheets/src/facade/f-range.ts [hunk 2]
```diff
@@ -66,6 +66,10 @@ export class FRange extends FBase {
         return this._range;
     }
 
+    override toString(withSheet?: boolean): string {
```
Comment (VicKun4937) [inline packages/sheets/src/facade/f-range.ts:69]: same to getA1Notation  ?

Comment (VicKun4937) [reply packages/sheets/src/facade/f-range.ts:69]: we should add coments for every facade api

File: packages/sheets-data-validation/src/facade/f-data-validation.ts [hunk 3]
```diff
@@ -14,20 +14,21 @@
...
 import { FDataValidationBuilder } from './f-data-validation-builder';
 
 export class FDataValidation {
     rule: IDataValidationRule;
-    private _worksheet: FWorksheet | undefined;
+    private _worksheet: Worksheet | undefined;
+    private _injector: Injector | undefined;
 
-    constructor(rule: IDataValidationRule, worksheet?: FWorksheet) {
+    constructor(rule: IDataValidationRule, worksheet?: Worksheet, _injector?: Injector) {
```
Comment (VicKun4937) [inline packages/sheets-data-validation/src/facade/f-data-validation.ts:30]: I find many code like following:
```
this.Injector!.xxx
````
why not move inject to first param and delete  optional "?"

**Review by VicKun4937 — APPROVED** [review]

**Pull request merged**